### ISSUE TITLE
Two interface with same identifier misbehavior

### ISSUE THREAD

**Issue by tinganho** [Microsoft/TypeScript#2224]
I just noticed that the compiler complains about this:
```typescript

interface Car {
	engine: string;
}

interface Car {
	wheels: string;
}

var myCar: Car = {
	wheels: 'Hej'
};

console.log(myCar);

```

```
Property 'engine' is missing in type '{ wheels: string; }'.
```
When I change to  :
```typescript
var myCar: Car = {
	engine: 'Hej'
};
```
It still complains
```
Property 'wheels' is missing in type '{ engine: string; }'.
```
Here is playground link:
http://www.typescriptlang.org/Playground#src=%0D%0Ainterface%20Car%20%7B%0D%0A%09engine%3A%20string%3B%0D%0A%7D%0D%0A%0D%0Ainterface%20Car%20%7B%0D%0A%09wheels%3A%20string%3B%0D%0A%7D%0D%0A%0D%0Avar%20myCar%3A%20Car%20%3D%20%7B%0D%0A%09wheels%3A%20'Hej'%0D%0A%7D%3B%0D%0A%0D%0Aconsole.log(myCar)%3B%0D%0A

I think it should complain that two interfaces can't use the same identifier in same scope/module?

**Comment by basarat** [comment]
-1 this is by design and useful for definition files.

**Comment by tinganho** [comment]
I think you are missing my point. If it where by design then it shouldn't issue any error in the first or second example? Now it is issue an error on both examples which is weird.

My suggestion I guess is just inline with other programming languages, where they complain if you use two declaration with the same identifier/name?

**Comment by basarat** [comment]
Typescript interfaces are "open ended". You effectively said "it needs both wheels and engines anything less and the compiler should complain. "

This allows you define stuff like query plugins and many other use cases for definition files.

**Comment by tinganho** [comment]
But I can define two props with?
```typescript
interface Car {
    wheels: string;
    engine: string;
}
```
or just extend the first one. Now I'm doing neither of these. I think people expect it to throw an error of "double declaration" ?

**Comment by NoelAbrahams** [comment]
This is called [declaration merging](https://github.com/Microsoft/TypeScript/wiki/Declaration%20Merging)

> But I can define two props 

What if you want to extend an existing interface provided by a library?

**Comment by tinganho** [comment]
@NoelAbrahams ok maybe it's feature rather than a bug.
